chrome/content/zotero/xpcom/data/search.js

```diff
@@ -942,10 +961,17 @@ Zotero.Search.idsToTempTable = Zotero.Promise.coroutine(function* (ids) {
 });
 
 
-/*
+/**
  * Build the SQL query for the search
+ *
+ * @param {String[]} parentKeys The stack of searches currently being executed
  */
-Zotero.Search.prototype._buildQuery = Zotero.Promise.coroutine(function* () {
+Zotero.Search.prototype._buildQuery = Zotero.Promise.coroutine(function* (parentKeys) {
+	if (this.key && parentKeys.includes(this.key)) {
+		Zotero.logError(new Error(`Search condition leads to loop (this: ${this.key}, parents: ${parentKeys})`));
+		return [];
```

Comment:
Sure, see added test. It's necessary because `_buildQuery()` calls `getSQL()` and `getSQLParams()` directly on subsearches without post-search filters, so we need some kind of check in either those methods or `_buildQuery()`. Putting it in `_buildQuery()` seemed cleanest.